### PR TITLE
Kan 23 text field with control

### PR DESCRIPTION
**Pull request by stasfilippov**

### CODE REVIEW

File: src/components/ui/textField/textField.tsx [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-  ComponentPropsWithoutRef,
-  useId,
-  useRef,
-  useState,
-} from 'react'
+import { ChangeEvent, ComponentPropsWithoutRef, forwardRef, useId, useRef, useState } from 'react'
 
 import { CloseOutline, EyeOffOutline, EyeOutline, SearchOutline } from '@/assets/icons/components'
+import { getType } from '@/components/ui/textField/utils/getTypes'
+import { mergeRefs } from '@/components/ui/textField/utils/merge-refs'
```
Comment (doroshkent) [inline src/components/ui/textField/textField.tsx:5]: it's better to create index file in utils and export from folder

File: src/components/ui/textField/textField.tsx [hunk 2]
```diff
@@ -16,122 +11,117 @@ import s from './textField.module.scss'
...
+    const finalRef = mergeRefs([forwardedRef, internalRef])
+    const isSearchField = type === 'search'
+    const isPasswordField = type === 'password'
 
-  const [isShowPassword, setIsShowPassword] = useState(false)
-  const [isActive, setIsActive] = useState(false)
-  const inputRef = useRef<HTMLInputElement>(null)
-  const isSearchField = type === 'search'
-  const isPasswordField = type === 'password'
+    const toggleShowPasswordHandler = () => setIsShowPassword(prevState => !prevState)
```
Comment (doroshkent) [inline src/components/ui/textField/textField.tsx:43]:
```suggestion
    const toggleShowPasswordHandler = setIsShowPassword(!isShowPassword)
```

File: src/components/ui/form/textField-form.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <TextFieldForm control={control} label={'Email'} name={'email'} />
+      <TextFieldForm control={control} label={'Password'} name={'password'} type={'password'} />
+      <Button type={'submit'}>Submit</Button>
+    </form>
+  )
+}
+
+export const Default: Story = {
+  render: () => <TextFieldFormStory />,
+}
```
Comment (doroshkent) [inline src/components/ui/form/textField-form.stories.tsx:46]: rm file

File: src/components/ui/textField/utils/getTypes.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+export const getType = (
+  type: ComponentProps<'input'>['type'],
+  isShowPassword: boolean
+): ComponentProps<'input'>['type'] => {
+  if (type === 'password' && isShowPassword) {
+    return 'text'
+  }
+
+  return type
+}
```
Comment (doroshkent) [inline src/components/ui/textField/utils/getTypes.ts:12]: do we indeed need separate func for that? is it going to be used anywhere else?

File: src/components/ui/textField/textField.tsx [hunk 5]
```diff
@@ -16,122 +11,117 @@ import s from './textField.module.scss'
...
 
-  const toggleShowPasswordHandler = () => setIsShowPassword(prevState => !prevState)
+    const generatedId = useId()
+    const domainId = id ?? generatedId
+    const domainType = getType(type, isShowPassword)
```
Comment (doroshkent) [inline src/components/ui/textField/textField.tsx:47]:
```suggestion
   const domainType = (type === 'password' && isShowPassword) ? 'text' : type;
```

File: src/components/ui/textField/textField.tsx [hunk 6]
```diff
@@ -16,122 +11,117 @@ import s from './textField.module.scss'
...
+          </Typography>
+        )}
+        <div className={clsx(s.fieldContainer, className)} {...rest}>
+          {isSearchField && (
+            <SearchOutline
+              className={clsx(
+                s.searchIcon,
+                isActive && s.searchIconActive,
+                disabled && s.iconDisabled
+              )}
```
Comment (doroshkent) [inline src/components/ui/textField/textField.tsx:83]: suggesting refactoring: create classNames obj

**Review by VladZimin — APPROVED** [review]

**Pull request merged**